initrd/bin/tpmr

```diff
@@ -192,86 +158,61 @@ coreboot TPM log:
  PCR-2 27c4f1fa214480c8626397a15981ef3a9323717f SHA1 [FMAP: FMAP]
 EOF
 
-# Returns first word of log - TPM2, TCPA or coreboot
-get_log_format () {
-	echo "$1" | head -n 1 | cut -d " " -f 1
+# awk script to handle all of the above. Note this gets squashed to one line so
+# semicolons are required.
+AWK_PROG='
+BEGIN {
+	getline;
+	hash_regex="([a-fA-F0-9]{40,})";
+	if ($0 == "TPM2 log:") {
+		RS="\n[^[:space:]]";
```

Comment:
That's a clever way to treat each log entry as a single record, I will have to remember this :wink: :100: